tests/__init__.py

```diff
@@ -2,8 +2,21 @@
 # Copyright (c) 2019 Mellanox Technologies, Inc . All rights reserved. See COPYING file
 
 import importlib
+import signal
 import os
 
+from pyverbs.pyverbs_error import PyverbsError
+
+
+# Handler for timeouts to be used by the test methods
+def sig_handler(signum, frame):
+    raise PyverbsError('Timeout expired')
+
+
+# Register the above handler to catch SIGALRM
+signal.signal(signal.SIGALRM, sig_handler)
```

Comment:
Hum. How this actually works is really tricky, I'm actually surprised it is useful.

What long operation are you expecting to break out of with this? Generally would be cleaner to add timers directly to that than to rely on EINTR working properly.

Reply:
There's no specific long test / operation which this was made for.
Its purpose is to handle hangouts in traffic cases, where the test could hang due to test/driver bug or any other reason that led to packet(s) drop.
(P.S: timeout decorator is also implemented in PyPI and I think it's ok to use (as long as there are no nested methods nor threads) )

Reply:
I'm not convinced this works as you think, alarm is very tricky, and won't effect most verbs calls. If it were to effect them it would trigger a verbs exception throw from EINTR, not alarm. To make this work properly all system calls must check for EINTR and then call PyErr_CheckSignals(). This includes all the calls inside pyverbs cython code.

We may have this wrong anyhow, we should probably be using some error handling path that goes through PyErr_SetFromErrno() which handles this and more.

Reply:
I see the concern more clearly now.
Hm.. I guess we can drop this PR for now, I'll consider changing the infrastructure flow and the way we handle exceptions.